manifests/init.pp

```diff
@@ -338,6 +350,21 @@
   Variant[String[1],Boolean] $dex_uri = '/dex',
   Openondemand::Dex_config $dex_config = {},
 
+  # Mellon Configs
+  Stdlib::Absolutepath $mellon_dir = "${apache::httpd_dir}/mellon",
+  Blloean mellon_manage_metadata = true,
+  Optional[Hash] $mellon_default_config = {
+    'MellonSPPrivateKeyFile' => "${mellon_dir}/mellon.key",
+    'MellonSPCertFile' => "${mellon_dir}/mellon.cer",
+    'MellonSPMetadataFile' => "${mellon_dir}/mellon_metadata.xml",
+    'MellonIdPMetadataFile' => "${mellon_dir}/idp_metadata.xml",
+    'MellonEnable' => 'auth',
+    'MellonEndpointPath' => '/mellon',
+  },
+  Optional[Hash] $mellon_config = {},
+  # Merge default config with updated configs
+  Optional[Hash] $mellon_merged_config = merge($mellon_default_config, $mellon_config),
```

Comment:
Default config should be in the class but not a parameter.  The `mellon_config` is fine as parameter but `mellon_default_config` and this merge should be happening further down not as parameters.